File: app/src/main/java/com/polok/eubmanagement/base/BaseActivity.java
package com.polok.eubmanagement.base;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewbinding.ViewBinding;

import com.polok.eubmanagement.widget.PrimaryLoader;

public abstract class BaseActivity<VB extends ViewBinding> extends AppCompatActivity {
    protected VB binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = initViewBinding();
        setContentView(binding.getRoot());
        initOnCreateView(savedInstanceState);
    }

    protected abstract VB initViewBinding();

    protected abstract void initOnCreateView(Bundle savedInstanceState);

    @Override
    protected void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: app/src/main/java/com/polok/eubmanagement/base/BaseActivity.kt
package com.polok.eubmanagement.base

import android.os.Bundle
import android.view.LayoutInflater
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.viewbinding.ViewBinding
import com.polok.eubmanagement.util.SharedPref

abstract class BaseActivity<VB: ViewBinding>(
    viewBindingFactory: (LayoutInflater) -> VB
) : AppCompatActivity() {

    protected val binding: VB by lazy {
        viewBindingFactory(layoutInflater)
    }
    protected var navController: NavController? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        /*window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_HIDDEN)
        if (Build.VERSION.SDK_INT < 30) window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
        else window?.setDecorFitsSystemWindows(true)*/
        setContentView(binding.root)
        SharedPref.init(this)
        initOnCreate(savedInstanceState)
    }

    abstract fun initOnCreate(savedInstanceState: Bundle?)
}